### DISCUSSION TITLE
Does the @golevelup/nestjs-stripe package work with Fastify?

### DISCUSSION THREAD

**Discussion by mhamill2** [golevelup/nestjs#890] [Q&A] [answered]
I'm trying to build an API using NestJS w/ Fastify (rather than Express), and I'm having some trouble getting it working. Currently running into the following error:

```
ERROR [ExceptionsHandler] No signatures found matching the expected signature for payload. Are you passing the raw request body you received from Stripe? 
 If a webhook request is being forwarded by a third-party tool, ensure that the exact request body, including JSON formatting and new line style, is preserved.
```

I have the `rawBody` argument set to true like the documentation shows, however I see the docs are using Express. I'm not sure if there's anything else that needs to be setup when using Fastify, so thought I'd ask here and also just double check that this package can indeed be used when using Fastify with NestJS rather than Express.

**Comment by underfisk** [comment, marked as answer]
We currently do not support Fastify, only express

**Comment by dtroox** [comment]
hey, @mhamill2!

in addition to setting `rawBody: true` in the NestJS bootstrap, you also need to configure `requestBodyProperty: "rawBody"` in the `StripeModule`:

```typescript
StripeModule.forRoot({
  webhookConfig: {
    requestBodyProperty: "rawBody",
  },
});
```

and if you’re testing webhooks with the Stripe CLI (`stripe listen`), add:

```typescript
StripeModule.forRoot({
  webhookConfig: {
    stripeSecrets: {
      accountTest: "whsec_...",
    },
    requestBodyProperty: "rawBody",
  },
})
```
